python_modules/dagster/dagster/_core/definitions/time_window_partitions.py

```diff
@@ -371,76 +394,46 @@ def get_current_timestamp(self, current_time: Optional[datetime] = None) -> floa
         None if the count cannot be computed quickly and must enumerate all partitions before
         counting them.
         """
-        last_partition_window = self.get_last_partition_window(current_time)
-        first_partition_window = self.get_first_partition_window(current_time)
-
-        if not last_partition_window or not first_partition_window:
-            return None
-
         if self.is_basic_daily:
             return (
                 date(
-                    last_partition_window.start.year,
-                    last_partition_window.start.month,
-                    last_partition_window.start.day,
+                    time_window.end.year,
+                    time_window.end.month,
+                    time_window.end.day,
                 )
                 - date(
-                    first_partition_window.start.year,
-                    first_partition_window.start.month,
-                    first_partition_window.start.day,
+                    time_window.start.year,
+                    time_window.start.month,
+                    time_window.start.day,
                 )
-            ).days + 1
+            ).days
 
         fixed_minute_interval = get_fixed_minute_interval(self.cron_schedule)
         if fixed_minute_interval:
-            minutes_in_window = (
-                last_partition_window.start.timestamp() - first_partition_window.start.timestamp()
-            ) / 60
-            return int(minutes_in_window // fixed_minute_interval) + 1
+            minutes_in_window = (time_window.end.timestamp() - time_window.start.timestamp()) / 60
+            return int(minutes_in_window // fixed_minute_interval)
 
-        return None
+        return len(self.get_partition_keys_in_time_window(time_window))
 
     def get_num_partitions(
         self,
         current_time: Optional[datetime] = None,
         dynamic_partitions_store: Optional[DynamicPartitionsStore] = None,
     ) -> int:
-        # Method added for performance reasons.
-        # Fetching partition keys requires significantly more compute time to
-        # string format datetimes.
-
-        fast_num_partitions = self._get_fast_num_partitions(current_time)
-        if fast_num_partitions is not None:
-            return fast_num_partitions
-
-        current_timestamp = self.get_current_timestamp(current_time=current_time)
-
-        partitions_past_current_time = 0
-
-        num_partitions = 0
-        for time_window in self._iterate_time_windows(self.start):
-            if self.end and time_window.end.timestamp() > self.end.timestamp():
-                break
-            if (
-                time_window.end.timestamp() <= current_timestamp
-                or partitions_past_current_time < self.end_offset
-            ):
-                num_partitions += 1
-
-                if time_window.end.timestamp() > current_timestamp:
-                    partitions_past_current_time += 1
-            else:
-                break
```

Comment:
i don't think there are any weird arcane cases I am leaving out by delegating to get_num_partitions_in_window?

Reply:
if there are, then we should fix them in the delegated method (but I don't think there are)